packages/extension/src/shared/transactions/index.ts

```diff
@@ -129,3 +132,25 @@ function buildGetTransactionsFn(metadataTransactions: Transaction[]) {
     )
   }
 }
+
+export async function pruneTransactionsInfiniteScrollCache() {
```

Comment:
Does this function try to clear the cache of unselected accounts?

Reply:
Yes, if we have a selected account (we normally have unless this is a new wallet), then it will clear the cache for infinite scroll transaction for all other accounts.

Reply:
If I switch accounts twice, does it mean all account cache for scrolling would be removed?

Reply:
If we switch account *and* close & open the extension wallet, it will clear all the account cache for transactions infinite scroll.

The consequence of not having the infinite scroll cache is that if we click the activity tab to look at transaction history, it will try to fetch from the explorer backend. 

Do you think this is too aggressive?

Reply:
Does SWR support LRU cache with a limited capacity? LRU cache can handle clearing automatically.

Reply:
SWR doesn't seem to have built-in support for LRU cache with limited capacity. I can investigate if we can integrate with 3rd party service to achieve this.